### PR TITLE
expose ff_multiplier argument of Transformer

### PR DESCRIPTION
**Pull request by kashif**
This PR exposes the `ff_multiplier` option of the Attention so that the UNet is able to set it.

cc @sayakpaul

### CODE REVIEW

**Review by sayakpaul — COMMENTED** [review]
Thanks!

Do we need to register this argument with `register_to_config()`?

**Comment by kashif** [comment]
I believe the wrapper function will register all the arguments to the config and since now `ff_multiplier` is there it should also register it.

File: src/diffusers/models/unet_2d_condition.py [hunk 1]
```diff
@@ -282,6 +284,7 @@ def __init__(
                 dual_cross_attention=dual_cross_attention,
                 use_linear_projection=use_linear_projection,
                 upcast_attention=upcast_attention,
+                ff_multiplier=ff_multiplier,
             )
         elif mid_block_type == "UNetMidBlock2DSimpleCrossAttn":
```
Comment (sayakpaul) [inline src/diffusers/models/unet_2d_condition.py:289]: Shouldn't we also include it in the case of `UNetMidBlock2DCrossAttn`?

**Review by patrickvonplaten — COMMENTED** [review]
What is the use case for this? I'm not too keen on exposing config params just to expose them

**Comment by kashif** [comment]
@patrickvonplaten no this was to make smaller student unets for the distillation experiments

File: src/diffusers/models/unet_2d_condition.py (hunk 1 above)
Comment (williamberman) [reply src/diffusers/models/unet_2d_condition.py:289]: The feed forward block is only when a full transformer is used as the attention mechanism. The simple cross attn block just uses a single attention block, so there's no feed forward block

Comment (kashif) [reply src/diffusers/models/unet_2d_condition.py:289]: ah i see...

**Comment by williamberman** [comment]
@kashif is the intention to only add the config to the feedforward of the middle block of the unet or to all down and up blocks that use a transformer for their cross attention mechanism? 

As is, I see we also add the config to a few other block in the unetblocks2d but because the ff_multiplier isn't passed through `get_{down,up}_block`, they won't receive the config.

**Comment by kashif** [comment]
for now, we wanted it for all attention feedforward blocks... but yes middle ones would also work i suppose. let me fix the `get_` helpers...

**Comment by patrickvonplaten** [comment]
I'd say we only merge this PR if we have a powerful distilled SD model - otherwise no need yet to expose more parameters IMO

**Comment by williamberman** [comment]
@kashif should we close this for now? I'm ok with revisiting in the future

**Comment by kashif** [comment]
yes please! i only needed it for my experiments. closing!

**Pull request closed without merging**